packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,445 @@
+      focusCurrentNode();
+      isPillDeleted.current = false;
+    }
+  }, [isPillDeleted, children]);
+
+  useEffect(() => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    // To focus the immediate focusable pill after the rollUp pill is selected/removed
+    if (isCollapsible && !isCollapsed) {
+      if (pills.length === React.Children.count(children)) {
+        setTabIndex('-1');
+        focusNode.current += 1;
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+        if (isRollUpRemoved.current) {
+          focusCurrentNode();
+          isRollUpRemoved.current = false;
+        }
+      }
+    }
+  }, [children, isCollapsed, isCollapsible, updatedCount]);
+
+  const focusNextNode = (pills, rollUpPill) => {
+    // if the next pill is roll up pill, focus the roll up pill
+
+    if (focusNode.current + 1 === pills.length || focusNode.current + 1 < pills.length) {
+      setTabIndex('-1');
+      if (rollUpPill && focusNode.current + 1 === pills.length) {
+        currentPill.current = rollUpPill.getAttribute('id');
+      } else if (focusNode.current + 1 < pills.length) {
+        focusNode.current += 1;
+        currentPill.current = children[focusNode.current].props.id;
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const focusPreviousNode = (pills, rollUpPill) => { // lp052179
+    // if the focused pill is roll up pill, then focus the immediate previous pill
+    if (rollUpPill && currentPill.current === 'rollup-pill') {
+      setTabIndex('-1');
+      focusNode.current = pills.length - 1;
+      currentPill.current = children[focusNode.current].props.id;
+      setTabIndex('0');
+      focusCurrentNode();
+    } else if (focusNode.current >= 1) { // focus the previous pill
+      setTabIndex('-1');
+      focusNode.current -= 1;
+      currentPill.current = children[focusNode.current].props.id;
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const focusNodeAfterDelete = (pills) => {
+    if (pills.length - 1 < React.Children.count(children)) {
+      setIsPillRemoved(true);
+      removedLabel.current = pills[focusNode.current].innerText;
+      setTabIndex('-1');
+      if (focusNode.current > 0) {
+        focusNode.current -= 1;
+      }
+      setUpdatedCount(React.Children.count(children));
+      isPillDeleted.current = true;
+    }
+  };
+
+  const handlePillListKeyDown = (event) => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = filterPillsRef.current.querySelector('[data-terra-rollup-pill]');
+    setIsPillRemoved(false);
+    setShowRollupPillInteraction(false);
+
+    switch (event.keyCode) {
+      case KEY_RIGHT:
+        event.preventDefault();
+        focusNextNode(pills, rollUpPill, false);
+        break;
+      case KEY_LEFT:
+        event.preventDefault();
+        focusPreviousNode(pills, rollUpPill);
```

Comment:
I can use the LEFT/RIGHT key to navigate the pills. However, I can use the Tab key to tab from one pill to the next. Tab only allows to set focus to the first pill. Is this behavior intended?

Reply:
It's expected behaviour. LEFT/RIGHT arrow should traverse the pills within the container. TAB key press is expected to Enter/Exit the pill container.